### PR TITLE
Rewrite in moon

### PR DESCRIPTION
**Pull request by sarahsturgeon**
Needs an actual test

### CODE REVIEW

File: lua/autorun/sh_waiter.moon [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+Waiter.lastLoopDuration = 0
+
+patronQueue = {}
+patronCount = 0
+
+-- Minimum delay (in seconds) between executions of attendPatrons()
+minBreakAfterAttending = 0.5
+
+generatePatron = (waitingFor, onSuccess, onTimeout, maxAttempts=0) ->
```
Comment (justinmuskopf) [inline lua/autorun/sh_waiter.moon:19]: Hello default args, where have you been my whole life?

File: lua/autorun/sh_waiter.moon [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    print "[CFC Waiter] Generating new Patron!"
+    patron =
+        :onSuccess,
+        :onTimeout,
+        :waitingFor,
+        :maxAttempts
+        attempts: 0
```
Comment (justinmuskopf) [inline lua/autorun/sh_waiter.moon:26]: coooooooooooooooooooooooooool

Comment (justinmuskopf) [reply lua/autorun/sh_waiter.moon:26]: Finger Guns

Comment (justinmuskopf) [reply lua/autorun/sh_waiter.moon:26]: hahahah embedded GIFS

File: lua/autorun/sh_waiter.moon [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+    patronQueue[patronCount] = patron
+
+    patronCount += 1
```
Comment (justinmuskopf) [inline lua/autorun/sh_waiter.moon:30]:
```suggestion
class PatronCount
    new: =>
        @count = 0
    increment: =>
        @count++
        
     patronCount = PatronCount!
     patronCount.increment()
```

;)

But really, why don't we `class` this bic tf up

File: lua/autorun/sh_waiter.moon [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+    print "[CFC Waiter] New Patron Count: #{patronCount}"
+
+removePatron = (patronID) ->
+    patronQueue[patronID] = nil
+
+Waiter.waitFor = (waitingFor, onSuccess, onTimeout) ->
+    print "[CFC Waiter] REgistering new Patron!"
```
Comment (justinmuskopf) [inline lua/autorun/sh_waiter.moon]:
```suggestion
    print "[CFC Waiter] Registering new Patron!"
```

File: lua/autorun/sh_waiter.moon [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+    timer.Simple delayTime attendPatrons
+
+getPatronsFromQueue = ->
+    print "[CFC Waiter] Retrieving patrons from WaiterQueue..."
+
+    for _, patron in pairs WaiterQueue
+        print "[CFC Waiter] Found patron in WaiterQueue! Importing..."
+
+        {:waitingFor, :onSuccess, :onTimeout, :maxAttempts} = patron
```
Comment (justinmuskopf) [inline lua/autorun/sh_waiter.moon:77]: What is going on with this Yoda shit man

**Review by justinmuskopf — CHANGES_REQUESTED** [review]
There's something about a print in here
I think we should really utilize the **moon** here, brother.
Make classes for these global objects. Merge and incorporate CFCLoggerLib.

File: lua/autorun/sh_waiter.moon (hunk 1 above)
Comment (sarahsturgeon) [reply lua/autorun/sh_waiter.moon:19]: they've been on the moon

File: lua/autorun/sh_waiter.moon [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+-- Minimum delay (in seconds) between executions of attendPatrons()
+minBreakAfterAttending = 0.5
+
+generatePatron = (waitingFor, onSuccess, onTimeout, maxAttempts=0) ->
+    print "[CFC Waiter] Generating new Patron!"
+    patron =
+        :onSuccess,
+        :onTimeout,
+        :waitingFor,
+        :maxAttempts
```
Comment (sarahsturgeon) [inline lua/autorun/sh_waiter.moon]:
```suggestion
        :maxAttempts,
```

File: lua/autorun/sh_waiter.moon (hunk 3 above)
Comment (sarahsturgeon) [reply lua/autorun/sh_waiter.moon:30]: I think we want everyone to use the same Waiter "instance" just given the way waiter is set up

File: lua/autorun/sh_waiter.moon (hunk 5 above)
Comment (sarahsturgeon) [reply lua/autorun/sh_waiter.moon:77]: _D E S T R U C T U R I N G_

File: lua/autorun/sh_waiter.moon (hunk 3 above)
Comment (justinmuskopf) [reply lua/autorun/sh_waiter.moon:30]: That there's called a `Singleton` class

File: lua/autorun/sh_waiter.moon [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+    for _, patron in pairs WaiterQueue
+        print "[CFC Waiter] Found patron in WaiterQueue! Importing..."
+
+        {:waitingFor, :onSuccess, :onTimeout, :maxAttempts} = patron
+
+        generatePatron waitingFor, onSuccess, onTimeout, maxAttempts
+
+getPatronsFromQueue!
+attendPatrons!
```
Comment (justinmuskopf) [inline lua/autorun/sh_waiter.moon:82]: lmao
GET PATRONS FROM QUEUE!
ATTEND PATRONS!

Comment (sarahsturgeon) [reply lua/autorun/sh_waiter.moon:82]: ATTEND!

**Comment by brandonsturgeon** [comment]
@justinmuskopf How about we test this, merge it, then rewrite with a more OO approach

**Comment by sarahsturgeon** [comment]
Closing this, we can reference back to this if we ever decide to dive down further with Moonscript

**Pull request closed without merging**